src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -202,80 +205,90 @@ ValidatorList::applyList (
 
     Json::Value list;
     PublicKey pubKey;
-    auto const result = verify (list, pubKey, manifest, blob, signature);
+    auto const result = verify(list, pubKey, manifest, blob, signature);
     if (result != ListDisposition::accepted)
         return result;
 
     // Update publisher's list
     Json::Value const& newList = list["validators"];
     publisherLists_[pubKey].available = true;
-    publisherLists_[pubKey].sequence = list["sequence"].asUInt ();
+    publisherLists_[pubKey].sequence = list["sequence"].asUInt();
     publisherLists_[pubKey].expiration = TimeKeeper::time_point{
         TimeKeeper::duration{list["expiration"].asUInt()}};
-    std::vector<PublicKey>& publisherList = publisherLists_[pubKey].list;
+    auto& publisherList = publisherLists_[pubKey].list;
 
-    std::vector<PublicKey> oldList = publisherList;
-    publisherList.clear ();
-    publisherList.reserve (newList.size ());
+    auto oldList = publisherList;
+    publisherList.clear();
+    publisherList.reserve(newList.size());
     std::vector<std::string> manifests;
     for (auto const& val : newList)
     {
-        if (val.isObject() &&
-            val.isMember ("validation_public_key") &&
-            val["validation_public_key"].isString ())
+        if (val.isObject())
         {
-            std::pair<Blob, bool> ret (strUnHex (
-                val["validation_public_key"].asString ()));
-
-            if (! ret.second || ! publicKeyType(makeSlice(ret.first)))
-            {
-                JLOG (j_.error()) <<
-                    "Invalid node identity: " <<
-                    val["validation_public_key"].asString ();
-            }
-            else
+            if (val.isMember("validation_public_key") &&
+                val["validation_public_key"].isString())
             {
-                publisherList.push_back (
-                    PublicKey(Slice{ ret.first.data (), ret.first.size() }));
+                std::pair<Blob, bool> ret(
+                    strUnHex(val["validation_public_key"].asString()));
+
+                if (!ret.second || !publicKeyType(makeSlice(ret.first)))
+                {
+                    JLOG (j_.error()) << "Invalid node identity: "
+                        << val["validation_public_key"].asString();
+                    continue;
+                }
+
+                std::string domain;
+
+                if (val.isMember(jss::domain) && val[jss::domain].isString())
+                    domain = val[jss::domain].asString();
+
+                publisherList.emplace_back(
+                    PublicKey{Slice{ret.first.data(), ret.first.size()}},
+                    domain);
             }
 
-            if (val.isMember ("manifest") && val["manifest"].isString ())
-                manifests.push_back(val["manifest"].asString ());
+            if (val.isMember("manifest") && val["manifest"].isString())
+                manifests.push_back(val["manifest"].asString());
```

Comment:
Is it an intentional change that this code now gets executed when `val` doesn't have a "validation_public_key", but before this code did not?

Reply:
I'm assuming it's intentional, and I think it's the right thing to do. We're interested in a valid manifest even if it's on an invalid/malformed list.